compiler/rustc_errors/src/lib.rs

```diff
@@ -925,7 +925,7 @@ impl Handler {
         self.inner.borrow_mut().emit_unused_externs(lint_level, unused_externs)
     }
 
-    pub fn delay_as_bug(&self, diagnostic: Diagnostic) {
+    pub fn delay_as_bug(&self, diagnostic: Diagnostic) -> Option<ErrorReported> {
```

Comment:
Shouldn't this be able to always return `ErrorReported`? Either way compilation will fail (the name of `ErrorReported` comes into here again I guess).

Reply:
@eddyb I originally thought so too, but `delay_as_bug` can check if `self.quiet` is true to see if an error should be emitted. I don't know when that would be true, but then no error is emitted, right?

Reply:
`self.quiet` is unsound and should be reverted (see other comment).

Reply:
Opened #93229